### DISCUSSION TITLE
Popover: Incluir un Textlink en el componente

### DISCUSSION THREAD

**Discussion by eziomenutti** [Telefonica/mistica-design#675] [Proposals]
Team: Metaverso

---

Los Popovers son super utiles para dar info contextual pero a veces el espacio con el que se cuenta es muy escaso para transmitir todo lo que se necesita. Por eso proponemos posibilitar la inclusión de un Textlink ampliando las posibilidades de comunicación del componente.

Algunos casos de uso:
- Popover que brinda más info acerca de un elemento y un CTA que lleva a una página de ayuda con el texto completo.
- Popover que sugiere modificar el valor de un campo acompañado de un CTA que realize la acción.

Imagen de ejemplo:
![Figma_ksQMmCiN1J](https://user-images.githubusercontent.com/46533404/183403084-3ac9bc9b-4c05-4c9e-a00c-98d811643464.png)

Muchas gracias!

**Comment by aweell** [comment]
Buenas @eziomenutti! Gracias por compartir la propuesta :blush:

Aquí añado mi granito de arena:

Una de las principales diferencias entre un tooltip y un popover a pesar de que los dos se presentan como un dialog no modal, es el tipo de contenido que incluyen, siendo el tooltip lo adecuado para un contenido no interactivo e informativo y el popover para un contenido interactivo.

En cuanto al comportamiento de su trigger, el tooltip se presenta tras hacer `:hover` en el trigger mientras que el popover puede hacerlo con `:hover` o con en el `onClick`.

Centrándome en el popover:

1. Totalmente de acuerdo en la posibilidad de incluir elementos interactivos en el componente
2. Dicho esto, no limitaría el tipo de contenido interactivo que puede incluirse en el mismo planteando que sólo sea un textLink, sino que permitiría otro tipo de elementos (Botones, inputs, imágenes...)

[Aquí un ejemplo de uso de un popover con inputFields](https://www.radix-ui.com/docs/primitives/components/popover)

Sin salir de esta misma discussion, GitHub utiliza popovers para las acciones relativas a usuarios:

![Screenshot 2022-08-08 at 13 43 10](https://user-images.githubusercontent.com/44420072/183410555-a6a65fe5-f3ed-4f2b-afba-6580dbb73397.png)

En resumen plantearía:

Un elemento estructuralmente flexible, en el que venga dado el modo de presentación de su contenedor, el control de dirección de aparición respecto al trigger y el control de colisiones.

**Reply by eziomenutti** [reply]
Sí, totalmente. La tuya es una propuesta superadora.

Nosotros en principio centramos nuestro pedido en sólo un textlink porque es una necesidad muy concreta que tenemos en el proyecto y nos interesa que, de ser considerada útil y prioritaria, que se desarrolle lo antes posible :)

**Reply by yceballost** [reply]
Completamente de acuerdo con el planteamiento.

Cómo abordarías esta evolución para evitar breaking changes? Añadirías a lo actual un Slot? Así podrías tener un popover con esa estructura de icono + title + description y luego el slot para que puedas montarte lo que quieras y no tendrías por qué utilizar los otros elementos.

**Reply by eziomenutti** [reply]
Yo agregaría un slot opcional con un Textlink a continuación del title y description.

Para sumar el resto de las cosas que mencionaste habria que analizar los casos de uso. Cualquier sugerencia que te haga al respecto ahora no va a tener en cuenta todas las necesidades en torno a esos nuevos elementos.

Se me ocurre definir ciertos campos relacionados a info que va sí o sí al principio del popover (Slot 1), otros elementos denominados "contenido" que son los que irian en el medio (Slot 2: Título, Imagen, descripción, bullet points) y otros elementos que siempre estarán al pie del popover como el CTA por ejemplo (Slot 3). Pero como te decía, no hice un análisis sobre este segundo punto y podría no ser la unica solución.

**Reply by aweell** [reply]
> Cómo abordarías esta evolución para evitar breaking changes? Añadirías a lo actual un Slot? Así podrías tener un popover con esa estructura de icono + title + description y luego el slot para que puedas montarte lo que quieras y no tendrías por qué utilizar los otros elementos.

@yceballost justo era en los que estaba pensando.

> Se me ocurre definir ciertos campos relacionados a info que va sí o sí al principio del popover (Slot 1), otros elementos denominados "contenido" que son los que irian en el medio (Slot 2: Título, Imagen, descripción, bullet points) y otros elementos que siempre estarán al pie del popover como el CTA por ejemplo (Slot 3). Pero como te decía, no hice un análisis sobre este segundo punto y podría no ser la unica solución.

@eziomenutti añadiendo únicamente un slot y teniendo los elementos actuales (title, description y asset) como opcionales, podría hacerse prácticamente cualquier cosa que fuese necesaria: incluir una lista, un avatar con datos de usuario, botones de acciones y por supuesto, un textlink 😊

**Reply by eziomenutti** [reply]
Genial.

No se cual es la timeline para esto, de nuestra parte hay mucha especulación para que este cambio se implemente rápido pero obviamente ustedes nos dirán los próximos pasos, los tiempos y de ser necesario sugerirnos un plan B para mientras tanto.

**Reply by yceballost** [reply]
Los tiempos dependen del backlog que tiene el equipo.(no podemos hacer todas las peticiones que tenemos en poco tiempo jaja) Siempre podrá ser más ágil si el cambio lo empuja el equipo que lo necesita (en este caso vosotros). Mística es un sistema colaborativo y cualquier equipo puede hacer pull requests :) 

https://brandfactory.telefonica.com/document/1846#/contribute/how-to-contribute
